### ISSUE TITLE
Socket Blocker in tests may break tests that use multiprocessing

### ISSUE THREAD

**Issue by keflavich** [astropy/astropy#2557]
Python's `multiprocessing` module uses `socket` to make connections between processes: https://github.com/python-git/python/blob/715a6e5035bb21ac49382772076ec4c630d6e960/Lib/multiprocessing/connection.py#L211

As a result, tests written using multiprocessing may fail with the usual error:
```
IOError: An attempt was made to connect to the internet by a test that was not marked `remote_data`.
```

So far, this doesn't affect anything in astropy-core, but it would be good to search for a workaround.

**Comment by embray** [comment]
I thought I fixed this.

**Comment by embray** [comment]
Yeah, I fixed this in #3713.
